### PR TITLE
fix: failing of continue-compilation flag for `int_01.f90` file

### PR DESCRIPTION
**Pull request by parth121101**
Towards: #5463 #5036

### CODE REVIEW

File: tests/reference/asr-int_01-9bf3eb9.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
...
     "outfile": null,
     "outfile_hash": null,
-    "stdout": null,
-    "stdout_hash": null,
+    "stdout": "asr-int_01-9bf3eb9.stdout",
+    "stdout_hash": "0a8d67e6e5a2f7501a36ea0a71c13d64407d4aff433355b1facb4a10",
     "stderr": "asr-int_01-9bf3eb9.stderr",
     "stderr_hash": "5a627fb0f74fc80baf2d78276ff8db8d9ace30e87afcd8792bfaa150",
-    "returncode": 2
+    "returncode": 0
```
Comment (certik) [inline tests/reference/asr-int_01-9bf3eb9.json:12]: What is going on here --- why did it fail before and now it passes?

File: src/lfortran/semantics/ast_common_visitor.h [hunk 2]
```diff
@@ -6661,7 +6661,11 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 if( ASRUtils::IntrinsicElementalFunctionRegistry::is_intrinsic_function(var_name) ) {
                     const bool are_all_args_evaluated { ASRUtils::all_args_evaluated(args, true) };
                     fill_optional_kind_arg(var_name, args);
-                    scalar_kind_arg(var_name, args);
+                    try {
+                        scalar_kind_arg(var_name, args);
+                    } catch (const SemanticAbort& e) {
+                        return nullptr;
+                    }
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:6668]: This doesn't seem right, but maybe it is. Why cannot you let the SemanticAbort exception go all the way to the top?  Why does silencing it and returning "nullptr" is needed here?

File: src/lfortran/semantics/ast_common_visitor.h [hunk 3]
```diff
@@ -6661,7 +6661,11 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 if( ASRUtils::IntrinsicElementalFunctionRegistry::is_intrinsic_function(var_name) ) {
                     const bool are_all_args_evaluated { ASRUtils::all_args_evaluated(args, true) };
                     fill_optional_kind_arg(var_name, args);
-                    scalar_kind_arg(var_name, args);
+                    try {
+                        scalar_kind_arg(var_name, args);
+                    } catch (const SemanticAbort& e) {
+                        return nullptr;
```
Comment (Pranavchiku) [inline src/lfortran/semantics/ast_common_visitor.h:6667]:
```suggestion
                        tmp = nullptr;
                        return nullptr;
```

Comment (Pranavchiku) [reply src/lfortran/semantics/ast_common_visitor.h:6667]:
```diff
diff --git a/src/lfortran/semantics/ast_common_visitor.h b/src/lfortran/semantics/ast_common_visitor.h
index af6c8ef58..bd2f841a6 100644
--- a/src/lfortran/semantics/ast_common_visitor.h
+++ b/src/lfortran/semantics/ast_common_visitor.h
@@ -3183,7 +3183,12 @@ public:
                         }
                     }
                 } else if (s.m_initializer != nullptr) {
-                    this->visit_expr(*s.m_initializer);
+                    try {
+                        this->visit_expr(*s.m_initializer);
+                    } catch (const SemanticAbort& e) {
+                        if (!compiler_options.continue_compilation) throw e;
+                        return;
+                    } 
                     if (is_compile_time && AST::is_a<AST::ArrayInitializer_t>(*s.m_initializer)) {
                         AST::ArrayInitializer_t *temp_array =
                             AST::down_cast<AST::ArrayInitializer_t>(s.m_initializer);
@@ -6664,6 +6669,8 @@ public:
                     try {
                         scalar_kind_arg(var_name, args);
                     } catch (const SemanticAbort& e) {
+                        tmp = nullptr;
+                        if (!compiler_options.continue_compilation) throw e;
                         return nullptr;
                     }
                     ASRUtils::create_intrinsic_function create_func =
```

File: src/lfortran/semantics/ast_common_visitor.h (hunk 2 above)
Comment (Pranavchiku) [reply src/lfortran/semantics/ast_common_visitor.h:6668]: Yes, diff at https://github.com/lfortran/lfortran/pull/5562#discussion_r1868227007 does it

File: src/lfortran/semantics/ast_common_visitor.h (hunk 3 above)
Comment (certik) [reply src/lfortran/semantics/ast_common_visitor.h:6667]: Can you simply set "tmp = nullptr" before `scalar_kind_arg(var_name, args)` and get rid of the try/catch?

**Pull request merged**